tests/test_testserver.py

```diff
@@ -135,3 +135,28 @@ def test_request_recovery_with_bigger_timeout(self):
             sock.close()
             
         assert server.handler_results[0] == data
+
+    def test_server_finishes_on_error(self):
+        """the server thread exits even if an exception exits the context manager"""
+        server = Server.basic_response_server()
+        try:
+            with server:
+                raise Exception()
+        except Exception:
+            pass
```

Comment:
Can we assert that the raise makes it out? `with pytest.raises()` would do that, rather than a `try...except`.